mitodl/edx-sga
# Python312 upgrade

#### What are the relevant tickets?
#356 

#### What's this PR do?
Upgrade Python to version 3.12 and remove all depreciation warnings 

#### How should this be manually tested?
- Check the CI run logs
- For local testing
    - Make sure you have python 3.12 installed on your system.
    - Run the tests using tox which has been configured to run them with python version 3.12

#### Note

~~The [xblock-sdk](https://github.com/openedx/xblock-sdk) workbench currently doesn't work with XBlock==2.0.0. Hopefully the issue shall be solved in the new release of xblock-sdk. Therefore, the version of xblock has been temporarily set to 1.10.0 in the [test_requirements.txt](https://github.com/mitodl/edx-sga/blob/17c80c7f85af6970f947ab8b4024b4a75a03e553/test_requirements.txt#L29)~~ Fixed
